src/modules/Login/action.js

```diff
@@ -176,6 +190,53 @@ export const initializeAccount = (account: EdgeAccount, touchIdInfo: Object) =>
   }
 }
 
+export const mergeSettings = (
+  loadedSettings: Object,
+  defaults: Object,
+  types: Object
+): { finalSettings: Object, isOverwriteNeeded: boolean, isDefaultTypeIncorrect: boolean } => {
+  const finalSettings = {}
+  // begin process for repairing damaged settings data
+  let isOverwriteNeeded = false
+  let isDefaultTypeIncorrect = false
+  for (const key in defaults) {
+    // if the type of the setting default does not meet the enforced type
+    const defaultSettingType = typeof defaults[key]
+    if (defaultSettingType !== types[key]) {
+      isDefaultTypeIncorrect = true
+      console.error('MismatchedDefaultSettingType key: ', key, ' with defaultSettingType: ', defaultSettingType, ' and necessary type: ', types[key])
```

Comment:
Should just throw here. This should be considered a coding error and be caught immediately

Reply:
Ok fair enough. I didn’t see the try catch.